test/unit/report_importer_test.rb

```diff
@@ -19,54 +19,4 @@ class ReportImporterTest < ActiveSupport::TestCase
     r = Report.import read_json_fixture('report-no-logs.json')
     assert_empty r.logs
   end
-
-  context 'with user owner' do
-    setup do
-      @owner = as_admin { FactoryGirl.create(:user, :admin, :with_mail) }
-      @host = FactoryGirl.create(:host, :owner => @owner)
-    end
-
-    test 'when owner is subscribed to notification, a mail should be sent on error' do
-      @owner.mail_notifications << MailNotification[:puppet_error_state]
-      assert_difference 'ActionMailer::Base.deliveries.size' do
-        report = read_json_fixture('report-errors.json')
-        report["host"] = @host.name
-        ReportImporter.import report
-      end
-    end
-
-    test 'when owner is not subscribed to notifications, no mail should be sent on error' do
-      @owner.mail_notifications = []
-      assert_no_difference 'ActionMailer::Base.deliveries.size' do
-        report = read_json_fixture('report-errors.json')
-        report["host"] = @host.name
-        ReportImporter.import report
-      end
-    end
-  end
-
-  test 'when a host has no owner, no mail should be sent on error' do
```

Comment:
reverted the tests. However, two tests are testing if mail is sent on puppet error state.
In those two tests I am using `ConfigReportImporter`
https://github.com/theforeman/foreman/pull/2750/files#diff-0e5400bfc3caabc55745ba764d2b9ecdR35 & https://github.com/theforeman/foreman/pull/2750/files#diff-0e5400bfc3caabc55745ba764d2b9ecdR65